hi-ml-histopathology/src/histopathology/utils/output_utils.py

```diff
@@ -344,35 +382,49 @@ def _save_outputs(self, epoch_results: EpochResultsType, metrics_dict: Mapping[M
 
         save_scores_histogram(results, figures_dir=figures_dir)
 
-        conf_matrix: ConfusionMatrix = metrics_dict[MetricsKey.CONF_MATRIX]  # type: ignore
-        save_confusion_matrix(conf_matrix, class_names=self.class_names, figures_dir=figures_dir)
+        # TODO: Re-enable plotting confusion matrix without relying on metrics to avoid DDP deadlocks
+        # conf_matrix: ConfusionMatrix = metrics_dict[MetricsKey.CONF_MATRIX]  # type: ignore
```

Comment:
so it is not possible to do confusion matrices after your change?

Reply:
It will be possible to recompute it from the `epoch_results`. The issue here was that calling `compute()` on the `ConfusioMatrix` metric triggers an `all_gather()` call; but because this is running only on rank 0, the code hangs waiting for the other processes.